torch/nn/quantized/modules/linear.py

```diff
@@ -110,14 +111,56 @@ def __init__(self, in_features, out_features, bias_=True):
         qweight = torch._empty_affine_quantized(
             [out_features, in_features], scale=1, zero_point=0, dtype=torch.qint8)
 
-        self._packed_weight = torch.ops.quantized.fbgemm_linear_prepack(qweight)
+        self.set_weight(qweight)
         self.scale = 1.0
         self.zero_point = 0
 
     def forward(self, x):
         return torch.ops.quantized.fbgemm_linear(
             x, self._packed_weight, self.bias, self.scale, self.zero_point)
 
+    # ===== Serialization methods =====
+    # The special consideration here is that we have to unpack the weights into their
+    # regular QTensor form for serialization. Packed weights should not live
+    # outside the process in which they were created, rather they should be derived
+    # from the QTensor weight.
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
+        super(Linear, self)._save_to_state_dict(destination, prefix, keep_vars)
+        destination[prefix + 'weight'] = self.weight()
+        destination[prefix + 'scale'] = torch.tensor(self.scale)
+        destination[prefix + 'zero_point'] = torch.tensor(self.zero_point)
+        if self.bias is not None:
+            destination[prefix + 'bias'] = self.bias
+
+    # ===== Deserialization methods =====
+    # Counterpart to the serialization methods, we must pack the serialized QTensor
+    # weight into its packed format for use by the FBGEMM ops.
+    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
+                              missing_keys, unexpected_keys, error_msgs):
+        self.set_weight(state_dict[prefix + 'weight'])
+        state_dict.pop(prefix + 'weight')
+
+        if prefix + 'bias' in state_dict:
+            self.bias.copy_(state_dict[prefix + 'bias'])
+            state_dict.pop(prefix + 'bias')
+
+        self.scale = float(state_dict[prefix + 'scale'])
+        state_dict.pop(prefix + 'scale')
+
+        self.zero_point = int(state_dict[prefix + 'zero_point'])
+        state_dict.pop(prefix + 'zero_point')
+
+        super()._load_from_state_dict(state_dict, prefix, local_metadata, False,
```

Comment:
..another one here.

Reply:
I'm surprised flake8 doesn't catch these in the pre-commit hook...